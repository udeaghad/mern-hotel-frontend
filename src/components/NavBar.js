import React from 'react';
import { NavLink } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { useSelector, useDispatch } from 'react-redux';
import axios from 'axios';
import AccountCircle from '@mui/icons-material/AccountCircle';
import getUser from '../redux/auths/usersAction';

const NavBar = () => {
  const user = useSelector((state) => state.user);
  const dispatch = useDispatch();

  const handleSignOut = async () => {
    localStorage.removeItem('user');

    dispatch(getUser(null));
    try {
      await axios.post('http://localhost:5000/api/v1/auths/logout', { withCredentials: true });
    } catch (error) {
      throw new Error(error.message);
    }
  };

  return (
    <Navbar bg="primary" expand="lg" variant="dark">
      <Container>
        <Navbar.Brand href="/">BOOooKa.com</Navbar.Brand>
        <AccountCircle />
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/">Home</NavLink>
            <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/reservations">Reservations</NavLink>
            <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/createhotel">Add Hotel</NavLink>
            <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/createroom">Add Room</NavLink>
            {!user
              ? (
                <>
                  <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/SignUp">Sign Up</NavLink>
                  <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/SignIn">Sign In</NavLink>
                </>
              )
              : (
                <>

                  <NavLink className="text-white text-decoration-none text-center py-2 mx-2" to="/" onClick={handleSignOut}>Sign Out</NavLink>
                </>
              )}
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default NavBar;
