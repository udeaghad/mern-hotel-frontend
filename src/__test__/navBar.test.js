import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/ConfigureStore';
import NavBar from '../components/NavBar';

describe('To test the Nav Bar component', () => {
  test('To test Home page renders successfully!', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <NavBar />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON;
    expect(tree).toMatchSnapshot();
  });

  test('Should display the navbar text text', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </BrowserRouter>,
    );
    const text = getByTestId('nav-bar');
    expect(text.textContent).toContain('Home');
    expect(text.textContent).toContain('Reservation');
    expect(text.textContent).toContain('Add Hotel');
    expect(text.textContent).toContain('Add Room');
    expect(text.textContent).toContain('Sign Up');
    expect(text.textContent).toContain('Sign In');
    expect(text.textContent).toContain('BOOooKa.com');
  });
});
