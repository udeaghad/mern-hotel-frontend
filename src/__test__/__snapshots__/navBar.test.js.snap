// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the Nav Bar component To test Home page renders successfully! 1`] = `[Function]`;
